src/Application/Migrations/15_function_ticket_state.sql

```diff
@@ -2,7 +2,7 @@ BEGIN;
 
 SET ROLE TO postgres;
 
-CREATE OR REPLACE FUNCTION ticket_state_next(param_project_id bigint, param_ticket_type enum_ticket_type, param_ticket_state enum_ticket_state)
+CREATE OR REPLACE FUNCTION ticket_state_next(param_ticket_id bigint)
```

Comment:
@jjeising @pegro 
[Continuing discussion](https://github.com/crs-tools/tracker/pull/162#discussion_r183407717)
This doesn't work, `Ticket::expandRecording` calls `Ticket::queryPreviousState` with `state='preparing'`, differing from the current state of the ticket, which would usually be `cutting`.
I'd suggest using an additional parameter `skip_dependent DEFAULT FALSE` and adding a new function (maybe with a trigger as suggested) `ticket_is_master` to pass that in from outside and keep this function completely ticket agnostic.

Reply:
As I said before, if determining the next state depends on more as just the ticket type and state, I'm fine with changing all functions to just requiring a ticket_id and migrate all users accordingly.

Reply:
>  I'm fine with changing all functions to just requiring a ticket_id

The new issue here is: ticket_state_next is also used with a state different from the current state of the ticket (with `param_ticket_id`).

Reply:
If you want to query for all possible ticket states of the current ticket, I would then add a second optional parameter param_state as a state of reference which defaults to the current ticket state.

So you could call ticket_state_next(param_ticket_id) to get the next state in relation to the current state. And you could call ticket_state_next(param_ticket_id, param_ticket_state) to query for the next state the ticket would get advanced to, **if** the ticket would be in state _param_ticket_state_.

At least that's what I'd suggest.

Reply:
@zuntrax That's what we discussed previously, can you implement that?